## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 553444`
Commit: `13e017662197fb96d0bb2cf74254008aaa287211`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python313Packages.bqplot</li>
    <li>python313Packages.bqscales</li>
    <li>python313Packages.ipydatagrid</li>
    <li>python314Packages.bqplot</li>
    <li>python314Packages.bqscales</li>
    <li>python314Packages.ipydatagrid</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python313Packages.bqplot</li>
    <li>python313Packages.bqscales</li>
    <li>python313Packages.ipydatagrid</li>
    <li>python314Packages.bqplot</li>
    <li>python314Packages.bqscales</li>
    <li>python314Packages.ipydatagrid</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:x: 25 packages failed to build:</summary>
  <ul>
    <li>python313Packages.deepspeed</li>
    <li>python313Packages.emmiai-noether</li>
    <li>python313Packages.ipympl</li>
    <li>python313Packages.jupyterlab-git</li>
    <li>python313Packages.jupyterlab-git-core</li>
    <li>python313Packages.mpltoolbox</li>
    <li>python313Packages.otter-grader</li>
    <li>python313Packages.plopp</li>
    <li>python313Packages.scippneutron</li>
    <li>python313Packages.spsdk</li>
    <li>python313Packages.vega</li>
    <li>python314Packages.deepspeed</li>
    <li>python314Packages.emmiai-noether</li>
    <li>python314Packages.ipympl</li>
    <li>python314Packages.jupyter-themes</li>
    <li>python314Packages.jupyterlab-git</li>
    <li>python314Packages.jupyterlab-git-core</li>
    <li>python314Packages.mpltoolbox</li>
    <li>python314Packages.otter-grader</li>
    <li>python314Packages.plopp</li>
    <li>python314Packages.scippneutron</li>
    <li>python314Packages.spsdk</li>
    <li>python314Packages.vega</li>
    <li>sage</li>
    <li>sageWithDoc</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 195 packages built:</summary>
  <ul>
    <li>aider-chat-full</li>
    <li>aider-chat-with-browser</li>
    <li>ihaskell</li>
    <li>jupyter</li>
    <li>jupyter-all</li>
    <li>kaggle</li>
    <li>markitdown</li>
    <li>markitdown-mcp</li>
    <li>mlflow-server</li>
    <li>nbqa</li>
    <li>open-webui</li>
    <li>pocket-tts</li>
    <li>python313Packages.ansible-kernel</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz-plots</li>
    <li>python313Packages.augmax</li>
    <li>python313Packages.batchspawner</li>
    <li>python313Packages.bayesian-optimization</li>
    <li>python313Packages.beat-this</li>
    <li>python313Packages.bqplot</li>
    <li>python313Packages.bqscales</li>
    <li>python313Packages.cometx</li>
    <li>python313Packages.compressai</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.dockerspawner</li>
    <li>python313Packages.einops</li>
    <li>python313Packages.encodec</li>
    <li>python313Packages.etils</li>
    <li>python313Packages.euporie</li>
    <li>python313Packages.experiment-utilities</li>
    <li>python313Packages.extra-streamlit-components</li>
    <li>python313Packages.flash-linear-attention</li>
    <li>python313Packages.guidance</li>
    <li>python313Packages.guidance-stitch</li>
    <li>python313Packages.hyper-connections</li>
    <li>python313Packages.ipycanvas</li>
    <li>python313Packages.ipydatagrid</li>
    <li>python313Packages.ipylab</li>
    <li>python313Packages.ipytablewidgets</li>
    <li>python313Packages.jupysql</li>
    <li>python313Packages.jupysql-plugin</li>
    <li>python313Packages.jupyter</li>
    <li>python313Packages.jupyter-collaboration</li>
    <li>python313Packages.jupyter-contrib-core</li>
    <li>python313Packages.jupyter-nbextensions-configurator</li>
    <li>python313Packages.jupyterhub</li>
    <li>python313Packages.jupyterhub-ldapauthenticator</li>
    <li>python313Packages.jupyterhub-systemdspawner</li>
    <li>python313Packages.jupyterhub-tmpauthenticator</li>
    <li>python313Packages.jupyterlab</li>
    <li>python313Packages.jupyterlab-execute-time</li>
    <li>python313Packages.jupyterlab-lsp</li>
    <li>python313Packages.jupyterlab-vim</li>
    <li>python313Packages.jupytext</li>
    <li>python313Packages.k-diffusion</li>
    <li>python313Packages.kaggle</li>
    <li>python313Packages.layoutparser</li>
    <li>python313Packages.lerobot</li>
    <li>python313Packages.llm-sentence-transformers</li>
    <li>python313Packages.local-attention</li>
    <li>python313Packages.markitdown</li>
    <li>python313Packages.mkdocs-jupyter</li>
    <li>python313Packages.mlcroissant</li>
    <li>python313Packages.mlflow</li>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mmengine</li>
    <li>python313Packages.nbdime</li>
    <li>python313Packages.nglview</li>
    <li>python313Packages.nikola</li>
    <li>python313Packages.nix-kernel</li>
    <li>python313Packages.notebook</li>
    <li>python313Packages.notedown</li>
    <li>python313Packages.oauthenticator</li>
    <li>python313Packages.orbax-checkpoint</li>
    <li>python313Packages.orbax-export</li>
    <li>python313Packages.pdfplumber</li>
    <li>python313Packages.pocket-tts</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pydeck</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymoo</li>
    <li>python313Packages.pytest-notebook</li>
    <li>python313Packages.pythreejs</li>
    <li>python313Packages.qgrid</li>
    <li>python313Packages.rotary-embedding-torch</li>
    <li>python313Packages.sagemaker-mlflow</li>
    <li>python313Packages.skops</li>
    <li>python313Packages.spandrel</li>
    <li>python313Packages.st-pages</li>
    <li>python313Packages.streamlit</li>
    <li>python313Packages.streamlit-avatar</li>
    <li>python313Packages.streamlit-card</li>
    <li>python313Packages.streamlit-echarts</li>
    <li>python313Packages.streamlit-folium</li>
    <li>python313Packages.streamlit-kpi-card</li>
    <li>python313Packages.streamlit-notify</li>
    <li>python313Packages.torch-einops-utils</li>
    <li>python313Packages.unstructured-inference</li>
    <li>python313Packages.vegafusion</li>
    <li>python313Packages.whisperx</li>
    <li>python313Packages.x-transformers</li>
    <li>python313Packages.xarray-einstats</li>
    <li>python314Packages.ansible-kernel</li>
    <li>python314Packages.arviz</li>
    <li>python314Packages.arviz-plots</li>
    <li>python314Packages.augmax</li>
    <li>python314Packages.batchspawner</li>
    <li>python314Packages.bayesian-optimization</li>
    <li>python314Packages.beat-this</li>
    <li>python314Packages.bqplot</li>
    <li>python314Packages.bqscales</li>
    <li>python314Packages.cometx</li>
    <li>python314Packages.compressai</li>
    <li>python314Packages.corner</li>
    <li>python314Packages.dockerspawner</li>
    <li>python314Packages.einops</li>
    <li>python314Packages.encodec</li>
    <li>python314Packages.experiment-utilities</li>
    <li>python314Packages.extra-streamlit-components</li>
    <li>python314Packages.flash-linear-attention</li>
    <li>python314Packages.guidance</li>
    <li>python314Packages.guidance-stitch</li>
    <li>python314Packages.hyper-connections</li>
    <li>python314Packages.ipycanvas</li>
    <li>python314Packages.ipydatagrid</li>
    <li>python314Packages.ipylab</li>
    <li>python314Packages.ipytablewidgets</li>
    <li>python314Packages.jupysql</li>
    <li>python314Packages.jupysql-plugin</li>
    <li>python314Packages.jupyter</li>
    <li>python314Packages.jupyter-collaboration</li>
    <li>python314Packages.jupyter-contrib-core</li>
    <li>python314Packages.jupyter-nbextensions-configurator</li>
    <li>python314Packages.jupyterhub</li>
    <li>python314Packages.jupyterhub-ldapauthenticator</li>
    <li>python314Packages.jupyterhub-systemdspawner</li>
    <li>python314Packages.jupyterhub-tmpauthenticator</li>
    <li>python314Packages.jupyterlab</li>
    <li>python314Packages.jupyterlab-execute-time</li>
    <li>python314Packages.jupyterlab-lsp</li>
    <li>python314Packages.jupyterlab-vim</li>
    <li>python314Packages.jupytext</li>
    <li>python314Packages.k-diffusion</li>
    <li>python314Packages.kaggle</li>
    <li>python314Packages.layoutparser</li>
    <li>python314Packages.lerobot</li>
    <li>python314Packages.llm-sentence-transformers</li>
    <li>python314Packages.local-attention</li>
    <li>python314Packages.markitdown</li>
    <li>python314Packages.mkdocs-jupyter</li>
    <li>python314Packages.mlflow</li>
    <li>python314Packages.mmcv</li>
    <li>python314Packages.mmengine</li>
    <li>python314Packages.nbdime</li>
    <li>python314Packages.nglview</li>
    <li>python314Packages.nikola</li>
    <li>python314Packages.nix-kernel</li>
    <li>python314Packages.notebook</li>
    <li>python314Packages.notedown</li>
    <li>python314Packages.oauthenticator</li>
    <li>python314Packages.pdfplumber</li>
    <li>python314Packages.pocket-tts</li>
    <li>python314Packages.pyannote-audio</li>
    <li>python314Packages.pydeck</li>
    <li>python314Packages.pymc</li>
    <li>python314Packages.pymoo</li>
    <li>python314Packages.pytest-notebook</li>
    <li>python314Packages.pythreejs</li>
    <li>python314Packages.qgrid</li>
    <li>python314Packages.rotary-embedding-torch</li>
    <li>python314Packages.sagemaker-mlflow</li>
    <li>python314Packages.skops</li>
    <li>python314Packages.spandrel</li>
    <li>python314Packages.st-pages</li>
    <li>python314Packages.streamlit</li>
    <li>python314Packages.streamlit-avatar</li>
    <li>python314Packages.streamlit-card</li>
    <li>python314Packages.streamlit-echarts</li>
    <li>python314Packages.streamlit-folium</li>
    <li>python314Packages.streamlit-kpi-card</li>
    <li>python314Packages.streamlit-notify</li>
    <li>python314Packages.torch-einops-utils</li>
    <li>python314Packages.unstructured-inference</li>
    <li>python314Packages.vegafusion</li>
    <li>python314Packages.whisperx</li>
    <li>python314Packages.x-transformers</li>
    <li>python314Packages.xarray-einstats</li>
    <li>quarto</li>
    <li>rstudio</li>
    <li>rstudioWrapper</li>
    <li>streamlit</li>
    <li>tocpdf</li>
    <li>unstructured-api</li>
    <li>whisper-ctranslate2</li>
    <li>whisperx</li>
  </ul>
</details>


---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.deepspeed</summary>
<pre>Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Using checkPhaseThreadLimitHook
Running phase: unpackPhase
unpacking source archive /nix/store/wg3156pnh8gwak7m55skgxk5yv6aczni-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file &quot;source/version.txt&quot;
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...

TIP pass --env-dir and --sdist-extract-dir to keep the build environment and sources, then see https://build.pypa.io/en/stable/how-to/troubleshooting.html#debug-a-failed-build for help debugging a failed build
ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel</pre>
</details>
<details>
<summary>python313Packages.ipympl</summary>
<pre>ERROR docs/examples/full-example.ipynb::Cell 2 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 3 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 4 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 5 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 6 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 7 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 8 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 9 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 10 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 11 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 12 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 13 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 14 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 15 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 16 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 17 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 18 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 19 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 20 - PermissionError: [Errno 1] Operation not permitted
=================== 8 passed, 8 warnings, 31 errors in 5.46s ===================</pre>
</details>
<details>
<summary>python313Packages.otter-grader</summary>
<pre>Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built otter_grader-6.1.6-py3-none-any.whl
Finished creating a wheel...
/nix/var/nix/builds/nix-58578-134096495/source/dist /nix/var/nix/builds/nix-58578-134096495/source
Unpacking to: unpacked/otter_grader-6.1.6...OK
Repacking wheel as ./otter_grader-6.1.6-py3-none-any.whl...OK
/nix/var/nix/builds/nix-58578-134096495/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for otter_grader-6.1.6-py3-none-any.whl
  - wrapt&lt;2.0.0,&gt;=1.16.0 not satisfied by version 2.2.2</pre>
</details>
<details>
<summary>python313Packages.spsdk</summary>
<pre>writing spsdk.egg-info/PKG-INFO
writing dependency_links to spsdk.egg-info/dependency_links.txt
writing entry points to spsdk.egg-info/entry_points.txt
writing requirements to spsdk.egg-info/requires.txt
writing top-level names to spsdk.egg-info/top_level.txt
writing manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;
/nix/store/rnp1jj0k7bifvyd5bx4xr3w2a878szhh-python3.13-vcs-versioning-1.1.1/lib/python3.13/site-packages/vcs_versioning/overrides.py:609: UserWarning: No GlobalOverrides context is active. Auto-creating one with SETUPTOOLS_SCM prefix for backwards compatibility. Consider using &#x27;with GlobalOverrides.from_env(&quot;YOUR_TOOL&quot;):&#x27; explicitly.
  return get_active_overrides().subprocess_timeout
[08/16/26 22:06:04] ERROR    listing git files failed - pretending    _git.py:37
                             there aren&#x27;t any                                   
reading manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;
reading manifest template &#x27;MANIFEST.in&#x27;
no previously-included directories found matching &#x27;.idea&#x27;
adding license file &#x27;LICENSE&#x27;
writing manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;

ERROR Unmet dependencies (checked against /nix/store/8gg2n5r87vsyy8d78n3s90a9r929c14v-python3-3.13.15/bin/python3.13):
        setuptools_scm&lt;10
                wanted: &lt;10
                found: 10.0.5</pre>
</details>
<details>
<summary>python313Packages.vega</summary>
<pre>                           &#x27;o&#x27;: pd.Series([np.array(i) for i in range(5)])})

vega/tests/test_utils.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/qalc8da55hw9898kncvkfljcq2jaf5yz-python3.13-pandas-3.0.4/lib/python3.13/site-packages/pandas/core/indexes/datetimes.py:1442: in date_range
    freq = to_offset(freq)
           ^^^^^^^^^^^^^^^
pandas/_libs/tslibs/offsets.pyx:6254: in pandas._libs.tslibs.offsets.to_offset
    ???
pandas/_libs/tslibs/offsets.pyx:6377: in pandas._libs.tslibs.offsets.to_offset
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

&gt;   ???
E   ValueError: Invalid frequency: H. Failed to parse with error message: ValueError(&quot;Invalid frequency: H. Failed to parse with error message: KeyError(&#x27;H&#x27;). Did you mean h?&quot;) Did you mean h?

pandas/_libs/tslibs/offsets.pyx:6162: ValueError
=========================== short test summary info ============================
FAILED vega/tests/test_utils.py::test_sanitize_dataframe - ValueError: Invalid frequency: H. Failed to parse with error message: Value...
========================= 1 failed, 4 passed in 0.53s ==========================</pre>
</details>
<details>
<summary>python314Packages.ipympl</summary>
<pre>ERROR docs/examples/full-example.ipynb::Cell 2 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 3 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 4 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 5 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 6 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 7 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 8 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 9 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 10 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 11 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 12 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 13 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 14 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 15 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 16 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 17 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 18 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 19 - PermissionError: [Errno 1] Operation not permitted
ERROR docs/examples/full-example.ipynb::Cell 20 - PermissionError: [Errno 1] Operation not permitted
=================== 8 passed, 8 warnings, 31 errors in 8.29s ===================</pre>
</details>
<details>
<summary>python314Packages.jupyter-themes</summary>
<pre>Traceback (most recent call last):
  File &quot;/nix/store/kmq6qdcg0yq6r4awf02dgq003n42r6zk-python3-3.14.7/lib/python3.14/importlib/metadata/__init__.py&quot;, line 407, in from_name
    return next(iter(cls.discover(name=name)))
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;&lt;string&gt;&quot;, line 1, in &lt;module&gt;
    from importlib.metadata import version; import sys; print(version(sys.argv[1]))
                                                              ~~~~~~~^^^^^^^^^^^^^
  File &quot;/nix/store/kmq6qdcg0yq6r4awf02dgq003n42r6zk-python3-3.14.7/lib/python3.14/importlib/metadata/__init__.py&quot;, line 987, in version
    return distribution(distribution_name).version
           ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/kmq6qdcg0yq6r4awf02dgq003n42r6zk-python3-3.14.7/lib/python3.14/importlib/metadata/__init__.py&quot;, line 960, in distribution
    return Distribution.from_name(distribution_name)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File &quot;/nix/store/kmq6qdcg0yq6r4awf02dgq003n42r6zk-python3-3.14.7/lib/python3.14/importlib/metadata/__init__.py&quot;, line 409, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for jupyter-themes</pre>
</details>
<details>
<summary>python314Packages.otter-grader</summary>
<pre>Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built otter_grader-6.1.6-py3-none-any.whl
Finished creating a wheel...
/nix/var/nix/builds/nix-6669-1511266414/source/dist /nix/var/nix/builds/nix-6669-1511266414/source
Unpacking to: unpacked/otter_grader-6.1.6...OK
Repacking wheel as ./otter_grader-6.1.6-py3-none-any.whl...OK
/nix/var/nix/builds/nix-6669-1511266414/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for otter_grader-6.1.6-py3-none-any.whl
  - wrapt&lt;2.0.0,&gt;=1.16.0 not satisfied by version 2.2.2</pre>
</details>
<details>
<summary>python314Packages.spsdk</summary>
<pre>writing spsdk.egg-info/PKG-INFO
writing dependency_links to spsdk.egg-info/dependency_links.txt
writing entry points to spsdk.egg-info/entry_points.txt
writing requirements to spsdk.egg-info/requires.txt
writing top-level names to spsdk.egg-info/top_level.txt
writing manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;
/nix/store/19wm7nlyv65ciwvanxq4sf14c4ww0v57-python3.14-vcs-versioning-1.1.1/lib/python3.14/site-packages/vcs_versioning/overrides.py:609: UserWarning: No GlobalOverrides context is active. Auto-creating one with SETUPTOOLS_SCM prefix for backwards compatibility. Consider using &#x27;with GlobalOverrides.from_env(&quot;YOUR_TOOL&quot;):&#x27; explicitly.
  return get_active_overrides().subprocess_timeout
[08/16/26 22:20:38] ERROR    listing git files failed - pretending    _git.py:37
                             there aren&#x27;t any                                   
reading manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;
reading manifest template &#x27;MANIFEST.in&#x27;
no previously-included directories found matching &#x27;.idea&#x27;
adding license file &#x27;LICENSE&#x27;
writing manifest file &#x27;spsdk.egg-info/SOURCES.txt&#x27;

ERROR Unmet dependencies (checked against /nix/store/kmq6qdcg0yq6r4awf02dgq003n42r6zk-python3-3.14.7/bin/python3.14):
        setuptools_scm&lt;10
                wanted: &lt;10
                found: 10.0.5</pre>
</details>
<details>
<summary>python314Packages.vega</summary>
<pre>pandas/_libs/tslibs/offsets.pyx:6377: in pandas._libs.tslibs.offsets.to_offset
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

&gt;   ???
E   ValueError: Invalid frequency: H. Failed to parse with error message: ValueError(&quot;Invalid frequency: H. Failed to parse with error message: KeyError(&#x27;H&#x27;). Did you mean h?&quot;) Did you mean h?

pandas/_libs/tslibs/offsets.pyx:6162: ValueError
=============================== warnings summary ===============================
vega/tests/test_outputs.py::test_vegalite_output
vega/tests/test_outputs.py::test_vegalite_output
vega/tests/test_outputs.py::test_vegalite_output
vega/tests/test_outputs.py::test_vega_output
  /nix/var/nix/builds/nix-32977-2859878243/vega-4.1.0/vega/utils.py:15: DeprecationWarning: codecs.open() is deprecated. Use open() instead.
    with codecs.open(abs_path(path), encoding=&#x27;utf-8&#x27;) as f:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED vega/tests/test_utils.py::test_sanitize_dataframe - ValueError: Invalid frequency: H. Failed to parse with error message: Value...
=================== 1 failed, 4 passed, 4 warnings in 0.77s ====================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/bqplot/default.nix
+++ b/pkgs/development/python-modules/bqplot/default.nix
@@ -1,38 +1,68 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  nodejs,
+  yarn-berry_3,
+
+  # build-system
   jupyter-packaging,
   jupyterlab,
+  setuptools,
+
+  # dependencies
   bqscales,
   ipywidgets,
   numpy,
   pandas,
   traitlets,
   traittypes,
+
+  # tests
+  nbval,
+  pytestCheckHook,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "bqplot";
-  version = "0.12.46";
+  version = "0.13.1";
   pyproject = true;
+  __structuredAttrs = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-lBdL4+skHJ+h1pelQxMomgvT/ogK3ZbGZA0xY73T0io=";
+  src = fetchFromGitHub {
+    owner = "bqplot";
+    repo = "bqplot";
+    tag = finalAttrs.version;
+    hash = "sha256-mYeQzmUa7WFDL8o6xsRD2bZ+3E9y4K/KyGE0zr0V4BA=";
   };
 
-  # upstream seems in flux for 0.13 release. they seem to want to migrate from
-  # jupyter_packaging to hatch, so let's patch instead of fixing upstream
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace "jupyter_packaging~=" "jupyter_packaging>=" \
-      --replace "jupyterlab~=" "jupyterlab>="
+  yarnOfflineCache = yarn-berry_3.fetchYarnBerryDeps {
+    src = "${finalAttrs.src}/js";
+    hash = "sha256-/QSZoanK92pV8g3n92g2pK17XjNgyerSe4UCV1/stXY=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    yarn-berry_3
+    yarn-berry_3.yarnBerryConfigHook
+  ];
+
+  # The yarn workspace lives in `js/`, so the offline install has to run from there.
+  dontYarnBerryInstallDeps = true;
+
+  preBuild = ''
+    pushd js
+    yarnBerryConfigHook
+    npm run build
+    popd
   '';
 
   build-system = [
     jupyter-packaging
     jupyterlab
+    setuptools
   ];
 
   dependencies = [
@@ -44,8 +74,12 @@ buildPythonPackage rec {
     traittypes
   ];
 
-  # no tests in PyPI dist, and not obvious to me how to build the js files from GitHub
-  doCheck = false;
+  nativeCheckInputs = [
+    nbval
+    pytestCheckHook
+  ];
+
+  enabledTestPaths = [ "tests/" ];
 
   pythonImportsCheck = [
     "bqplot"
@@ -55,7 +89,9 @@ buildPythonPackage rec {
   meta = {
     description = "2D plotting library for Jupyter based on Grammar of Graphics";
     homepage = "https://bqplot.github.io/bqplot";
+    downloadPage = "https://github.com/bqplot/bqplot";
+    changelog = "https://github.com/bqplot/bqplot/releases/tag/${finalAttrs.src.tag}";
     license = lib.licenses.asl20;
     maintainers = with lib.maintainers; [ bcdarwin ];
   };
-}
+})

--- a/pkgs/development/python-modules/bqscales/default.nix
+++ b/pkgs/development/python-modules/bqscales/default.nix
@@ -2,9 +2,8 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  python,
 
-  # frontend
+  # nativeBuildInputs
   nodejs,
   yarn-berry_3,
 
@@ -45,23 +44,13 @@ buildPythonPackage (finalAttrs: {
 
   nativeBuildInputs = [
     nodejs
-    yarn-berry_3.yarnBerryConfigHook
     yarn-berry_3
+    yarn-berry_3.yarnBerryConfigHook
   ];
 
-  preBuild =
-    # `jupyter labextension build` looks up `@jupyterlab/core-meta` in `node_modules` and
-    # downloads it from npm when missing. Seed it with the core package definition shipped
-    # by our `jupyterlab`, which is what the upstream `@jupyter/builder` devDependency
-    # would provide.
-    ''
-      mkdir -p node_modules/@jupyterlab/core-meta
-      cp ${jupyterlab}/${python.sitePackages}/jupyterlab/staging/package.json \
-        node_modules/@jupyterlab/core-meta/core.package.json
-    ''
-    + ''
-      npm run build
-    '';
+  preBuild = ''
+    npm run build
+  '';
 
   build-system = [
     hatch
@@ -81,7 +70,7 @@ buildPythonPackage (finalAttrs: {
 
   env.SKIP_JUPYTER_BUILDER = 1;
 
-  # no tests in PyPI dist
+  # no python tests
   doCheck = false;
 
   pythonImportsCheck = [ "bqscales" ];

--- a/pkgs/development/python-modules/ipydatagrid/default.nix
+++ b/pkgs/development/python-modules/ipydatagrid/default.nix
@@ -2,43 +2,57 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  pytestCheckHook,
 
+  # nativeBuildInputs
+  yarn-berry_3,
+
+  # build-system
   hatchling,
   hatch-jupyter-builder,
   jupyterlab,
+
+  # dependencies
   bqplot,
   ipywidgets,
   pandas,
   py2vega,
-  yarn-berry_3,
+
+  # tests
+  pytestCheckHook,
 }:
 
-let
-  yarn-berry = yarn-berry_3;
-in
-
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "ipydatagrid";
   version = "1.4.0";
   pyproject = true;
+  __structuredAttrs = true;
 
   src = fetchFromGitHub {
     owner = "jupyter-widgets";
     repo = "ipydatagrid";
-    tag = version;
+    tag = finalAttrs.version;
     hash = "sha256-6jaIYgLbNXIYzM+mZIVMZ1CXOpcbVK5k9nzGjq5UdLI=";
   };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml package.json \
+      --replace-fail 'jlpm' 'yarn'
+  '';
+
+  yarnOfflineCache = yarn-berry_3.fetchYarnBerryDeps {
+    inherit (finalAttrs) src;
+    hash = "sha256-5KZl9mK6xNvy2XdWieH20hEZJ+h/KzvjOfpo78FlWpg=";
+  };
+
+  nativeBuildInputs = [
+    yarn-berry_3
+    yarn-berry_3.yarnBerryConfigHook
+  ];
 
   build-system = [
     hatchling
     hatch-jupyter-builder
     jupyterlab
-  ];
-
-  nativeBuildInputs = [
-    yarn-berry.yarnBerryConfigHook
-    yarn-berry
   ];
 
   dependencies = [
@@ -48,25 +62,22 @@ buildPythonPackage rec {
     py2vega
   ];
 
-  offlineCache = yarn-berry.fetchYarnBerryDeps {
-    inherit src;
-    hash = "sha256-5KZl9mK6xNvy2XdWieH20hEZJ+h/KzvjOfpo78FlWpg=";
-  };
-
-  preConfigure = ''
-    substituteInPlace pyproject.toml package.json \
-      --replace-fail 'jlpm' 'yarn'
-  '';
+  pythonImportsCheck = [ "ipydatagrid" ];
 
   nativeCheckInputs = [
     pytestCheckHook
   ];
 
+  disabledTests = [
+    # Expects the pandas 2 table schema; pandas 3 tags string columns with `extDtype: "str"`
+    "test_data_object_generation"
+  ];
+
   meta = {
     description = "Fast Datagrid widget for the Jupyter Notebook and JupyterLab";
     homepage = "https://github.com/jupyter-widgets/ipydatagrid";
-    changelog = "https://github.com/jupyter-widgets/ipydatagrid/releases/tag/${version}";
+    changelog = "https://github.com/jupyter-widgets/ipydatagrid/releases/tag/${finalAttrs.src.tag}";
     license = lib.licenses.bsd3;
     maintainers = with lib.maintainers; [ flokli ];
   };
-}
+})

